### PR TITLE
add new placeholder tab header and footer

### PR DESCRIPTION
**Pull request by ImYenil**
add new placeholder tab header and footer

proof that works
image

### CODE REVIEW

**Comment by KKiiya** [comment]
Could you send a video and show that it actually increases and then you go to another game

**Comment by ImYenil** [comment]
> Could you send a video and show that it actually increases and then you go to another game

I noticed some bugs. I'll fix the bug and send it to you

**Comment by ImYenil** [comment]
> Could you send a video and show that it actually increases and then you go to another game

Fixed

File: bedwars-plugin/src/main/java/com/tomkeuper/bedwars/sidebar/BoardManager.java [hunk 1]
```diff
@@ -459,37 +460,40 @@ private void setHeaderFooter(TabPlayer player, IArena arena) {
...
                     player, lang.m(Messages.FORMATTING_SIDEBAR_TAB_HEADER_SPECTATOR),
-                    lang.m(Messages.FORMATTING_SIDEBAR_TAB_FOOTER_SPECTATOR)
+                    lang.m(Messages.FORMATTING_SIDEBAR_TAB_FOOTER_SPECTATOR).replace("%bw_kills%", String.valueOf(arena.getPlayerKills(((Player) player.getPlayer()).getPlayer(), false))).replace("%bw_final_kills%", String.valueOf(arena.getPlayerKills(((Player) player.getPlayer()).getPlayer(), true))).replace("%bw_beds%", String.valueOf(arena.getPlayerBedsDestroyed(((Player) player.getPlayer()).getPlayer())))
             );
             return;
         }
 
-        String headerPath = null;
-        String footerPath = null;
+        Bukkit.getServer().getScheduler().scheduleSyncRepeatingTask(plugin, () -> {
```
Comment (tomkeuper) [inline bedwars-plugin/src/main/java/com/tomkeuper/bedwars/sidebar/BoardManager.java:468]: Why?

File: bedwars-plugin/src/main/java/com/tomkeuper/bedwars/arena/Arena.java [hunk 2]
```diff
@@ -157,11 +157,11 @@ public class Arena implements IArena {
...
-    private HashMap<String, Integer> playerKills = new HashMap<>();
-    private HashMap<Player, Integer> playerBedsDestroyed = new HashMap<>();
-    private HashMap<Player, Integer> playerFinalKills = new HashMap<>();
-    private HashMap<Player, Integer> playerDeaths = new HashMap<>();
-    private HashMap<Player, Integer> playerFinalKillDeaths = new HashMap<>();
+    private final HashMap<String, Integer> playerKills = new HashMap<>();
+    private final HashMap<Player, Integer> playerBedsDestroyed = new HashMap<>();
+    private final HashMap<Player, Integer> playerFinalKills = new HashMap<>();
+    private final HashMap<Player, Integer> playerDeaths = new HashMap<>();
+    private final HashMap<Player, Integer> playerFinalKillDeaths = new HashMap<>();
```
Comment (tomkeuper) [inline bedwars-plugin/src/main/java/com/tomkeuper/bedwars/arena/Arena.java:164]: Whats the gain of changing this?

**Comment by tomkeuper** [comment]
And wouldnt it be better to pass the player to the Language#m() function and get the stats thru the api? this way we can prevent duplicated code

File: bedwars-plugin/src/main/java/com/tomkeuper/bedwars/arena/Arena.java (hunk 2 above)
Comment (ImYenil) [reply bedwars-plugin/src/main/java/com/tomkeuper/bedwars/arena/Arena.java:164]: > Whats the gain of changing this?



> Whats the gain of changing this?

There was an NPE bug when getting the value of this variable. I fixed this error by adding final code.

File: bedwars-plugin/src/main/java/com/tomkeuper/bedwars/sidebar/BoardManager.java (hunk 1 above)
Comment (ImYenil) [reply bedwars-plugin/src/main/java/com/tomkeuper/bedwars/sidebar/BoardManager.java:468]: > Why?

I need a repeat task to update this placeholder. But I haven't been able to fully fix it yet.

**Comment by tomkeuper** [comment]
Not the way to go

**Pull request closed without merging**